clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_cli.py

```diff
@@ -1,234 +1,154 @@
-            default=None,
-        )
-
-        optional_custom = self._args.add_argument_group(
-            "Pipeline options if you chose `custom` modality"
-        )
-        optional_custom.add_argument(
-            "-cn",
-            "--custom_suffix",
-            help="""Custom suffix filename, e.g.:
-                'graymatter_space-Ixi549Space_modulated-off_probability.nii.gz', or
-                'segm-whitematter_probability.nii.gz'
-                """,
-            type=str,
-            default="",
-        )
-
-        # Clinica standard arguments (e.g. --n_procs)
-        self.add_clinica_standard_arguments()
-
-    def run_command(self, args):
-        """Run the pipeline with defined args."""
-        from networkx import Graph
-
-        from clinica.utils.ux import print_crash_files_and_exit, print_end_pipeline
-
-        from .deeplearning_prepare_data_pipeline import DeepLearningPrepareData
-
-        parameters = {
-            # Add your own pipeline parameters here to use them inside your
-            # pipeline. See the file `deeplearning_prepare_data_pipeline.py` to
-            # see an example of use.
-            "modality": args.modality,
-            "extract_method": args.extract_method,
-            "patch_size": args.patch_size,
-            "stride_size": args.stride_size,
-            "slice_direction": args.slice_direction,
-            "slice_mode": args.slice_mode,
-            "roi_list": args.roi_list,
-            "roi_uncrop_output": args.roi_uncrop_output,
-            "custom_template": args.custom_template,
-            "custom_mask_pattern": args.custom_mask_pattern,
-            "use_uncropped_image": args.use_uncropped_image,
-            "custom_suffix": args.custom_suffix,
-            "acq_label": args.acq_label,
-            "suvr_reference_region": args.suvr_reference_region,
-        }
-
-        pipeline = DeepLearningPrepareData(
-            caps_directory=self.absolute_path(args.caps_directory),
-            tsv_file=self.absolute_path(args.subjects_sessions_tsv),
-            base_dir=self.absolute_path(args.working_directory),
-            parameters=parameters,
-            name=self.name,
-        )
-
-        if args.n_procs:
-            exec_pipeline = pipeline.run(
-                plugin="MultiProc", plugin_args={"n_procs": args.n_procs}
-            )
-        else:
-            exec_pipeline = pipeline.run()
-
-        if isinstance(exec_pipeline, Graph):
-            print_end_pipeline(
-                self.name, pipeline.base_dir, pipeline.base_dir_was_specified
-            )
-        else:
-            print_crash_files_and_exit(args.logname, pipeline.base_dir)
+from typing import Optional
+
+import click
+
+from clinica.pipelines import cli_param
+from clinica.utils.pet import LIST_SUVR_REFERENCE_REGIONS
+
+pipeline_name = "deeplearning-prepare-data"
+
+
+@click.command(name=pipeline_name)
+@cli_param.argument.caps_directory
+@click.argument(
```

Comment:
https://click.palletsprojects.com/en/8.0.x/documentation/#documenting-arguments

The standard CLI convention is to describe the arguments in the actual description of the command, not as a separate section below it. Although `argparse` enabled this because being in the Python stdlib it needs to be as generic as possible, the practice followed by Click and other modern CLI tools is saner:

- Arguments are mandatory so they should at the forefront of the CLI docs. When I run `click <command> -h` they should pop straight away in the short description at the top and be explained in the long description below.
- Options are for advanced usage and should have sane defaults already. Therefore, they are listed in a separate section below the short and long descriptions of the CLI doc string to not stand in the way.

Reply:
I see... It'k ok for me if the arguments are described in the description of the command, as proposed by Click in the link that you posted. In that case, it should be done for most of the `*_cli.py` files modified by this PR. For example, currently when I type `clinica run pet-linear --help`, I get this:
```
Usage: clinica run pet-linear [OPTIONS] BIDS_DIRECTORY CAPS_DIRECTORY

  Affine registration of PET images to the MNI standard space."

  See https://aramislab.paris.inria.fr/clinica/docs/public/latest/Pipelines/PE
  T_Linear/"

Options:
.
.
.
```
As user, I expect to have some minimal information about what `BIDS_DIRECTORY` and `CAPS_DIRECTORY` is. Something like this:
```
Usage: clinica run pet-linear [OPTIONS] BIDS_DIRECTORY CAPS_DIRECTORY

  Affine registration of PET images to the MNI standard space."
  
  BIDS_DIRECTORY is the path to the BIDS directory.
  CAPS_DIRECTORY is the path to the CAPS directory.

  See https://aramislab.paris.inria.fr/clinica/docs/public/latest/Pipelines/PE
  T_Linear/"

Options:
.
.
.
```
BTW, this pipeline has two other arguments that are not included (`ACQ_LABEL` and `SUVR`) and for those arguments a short description is definitely necessary.

Reply:
Agreed

Reply:
Although, one might find this description adds nothing of value because the naming of the argument is already quite explicit:

```
BIDS_DIRECTORY is the path to the BIDS directory.
CAPS_DIRECTORY is the path to the CAPS directory.
```

I think what the Click documentation says is that you should add some more contrext in the long description *using* the name of the arguments. So in this example, something along the lines of:

```
This pipeline performs registration of PET images found in the input BIDS dataset referenced by BIDS_DIRECTORY. The results of the registration are written to the ouput CAPS dataset referenced by CAPS_DIRECTORY. <plus paragraph about acq_label ad suvr>
```

I believe that's more along the lines of what is expected from a good CLI help string. Like what's in the description section of `man cp` for instance.